bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/ServerStreamingCallableWithMetrics.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc;
+
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.ResponseObserver;
+import com.google.api.gax.rpc.ServerStreamingCallable;
+import com.google.api.gax.rpc.StreamController;
+import com.google.cloud.bigtable.metrics.BigtableClientMetrics;
+import com.google.cloud.bigtable.metrics.Timer;
+import java.util.concurrent.TimeUnit;
+
+public class ServerStreamingCallableWithMetrics<RequestT, ResponseT>
+    extends ServerStreamingCallable<RequestT, ResponseT> {
+
+  private final ServerStreamingCallable<RequestT, ResponseT> streamingCallable;
+
+  private static Timer firstResponseTimer;
+  private Long startTime;
+
+  private static synchronized Timer getFirstResponseTimer() {
+    if (firstResponseTimer == null) {
+      firstResponseTimer =
+          BigtableClientMetrics.timer(
+              BigtableClientMetrics.MetricLevel.Info, "grpc.method.ReadRows.firstResponse.latency");
+    }
+    return firstResponseTimer;
+  }
+
+  ServerStreamingCallableWithMetrics(
+      ServerStreamingCallable<RequestT, ResponseT> streamingCallable) {
+    this.streamingCallable = streamingCallable;
+    if (BigtableClientMetrics.isEnabled(BigtableClientMetrics.MetricLevel.Info)) {
+      startTime = System.nanoTime();
+    }
+  }
+
+  @Override
+  public void call(
+      RequestT requestT,
+      final ResponseObserver<ResponseT> responseObserver,
+      ApiCallContext apiCallContext) {
+
+    final RpcMetricsV2 rpc = RpcMetricsV2.createRpcMetrics("readRows");
+    final Timer.Context rpcTimeOperation = rpc.timeOperation();
+
+    final ResponseObserver<ResponseT> wrapper =
+        new ResponseObserver<ResponseT>() {
+          @Override
+          public void onStart(StreamController streamController) {
+            responseObserver.onStart(streamController);
+          }
+
+          @Override
+          public void onResponse(ResponseT responseT) {
+            if (startTime != null) {
```

Comment:
Also, make sure that startTime is volatile since its used across different threads